internal/webhook/freight/webhook.go

```diff
@@ -211,6 +235,19 @@ func (w *webhook) ValidateUpdate(
 			},
 		)
 	}
+
+	req, err := w.admissionRequestFromContextFn(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("get admission request from context: %w", err)
+	}
+	// Record Freight approved events if the request doesn't come from Kargo controlplane.
+	if !w.isRequestFromKargoControlplaneFn(req) {
+		for approvedStage := range newFreight.Status.ApprovedFor {
+			if _, ok := oldFreight.Status.ApprovedFor[approvedStage]; !ok {
+				w.recordFreightApprovedEvent(req, newFreight, approvedStage)
+			}
+		}
```

Comment:
I think it's good that this is here and I wouldn't change it, but want to mention that I'm not sure if approval is actually possible except via API. I've never had luck mutating status directly from `kubectl`. iirc, there is some limitation to controller runtime that prevents it.

Reply:
This is _technically_ possible using `kubectl edit --subresource=status`.

Reply:
@krancour @hiddeco I found that I miss grant webhook-server to record event 😓
After granting it in https://github.com/akuity/kargo/pull/1739/commits/41165fe15de6a24f98c165f35dc4814c0962209c and as @hiddeco mentioned - using `kubectl edit --subresource=status` records event: